### PR TITLE
Add Unity build to the GHA pipeline

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

Fixes https://github.com/realm/realm-dotnet/issues/2285

* Replaces the code to unstash wrapper binaries with a loop
* Adds steps to build the unity package both with and without deps

### CODE REVIEW

File: .github/templates/main.yml [hunk 1]
```diff
@@ -79,6 +79,20 @@ steps:
...
+  - name: #@ "Build Unity" + (" Bundled Dependencies" if includeDependencies else "")
+    run: #@ "dotnet run --project Tools/SetupUnityPackage/ -- realm --packages-path Realm/packages --pack" + (" --include-dependencies" if includeDependencies else "")
+  - name: Store Unity artifacts
+    uses: actions/upload-artifact@v2
+    with:
+      name: #@ finalPkgName
+      path: #@ "${{ github.workspace }}/Realm/Realm.Unity/" + finalPkgName
+      retention-days: ${{ github.event_name != 'pull_request' && 30 || 1 }}
+  - name: Cleanup build folder
+    run: #@ "rm ${{ github.workspace }}/Realm/Realm.Unity/" + finalPkgName
```
Comment (LaPeste) [inline .github/templates/main.yml:93]: Why do you need to clean the build folder?
I understand that you call this function twice (line 267 and 268) but the final package of each call has a different name. What am I missing?

File: Jenkinsfile [hunk 2]
```diff
@@ -218,6 +218,7 @@ stage('Unity Package') {
   }
 }
 
+
```
Comment (LaPeste) [inline Jenkinsfile:221]: I believe this new line can go.

**Review by LaPeste — APPROVED** [review]
Good clean up thanks to the loop to unstash the binaries of the wrappers.

File: Jenkinsfile (hunk 2 above)
Comment (nirinchev) [reply Jenkinsfile:221]: Ah yeah, I wanted to trigger a build to test something 😅

**Pull request merged**